integration_tests/tests/common/mod.rs

```diff
@@ -143,14 +125,145 @@ pub async fn submit_measurements_and_verify_aggregate(
+            .unwrap(),
+    )
+    .await
+    .unwrap();
 
     // Verify that we got the correct result.
     assert_eq!(
-        collection,
-        Collection::new(total_measurements as u64, num_nonzero_measurements as u64)
+        collection.report_count(),
+        u64::try_from(measurements.len()).unwrap()
     );
+    assert_eq!(collection.aggregate_result(), aggregate_result);
+}
+
+pub async fn submit_measurements_and_verify_aggregate(
+    (leader_port, helper_port): (u16, u16),
+    leader_task: &Task,
+    collector_private_key: &HpkePrivateKey,
+) {
+    // Translate aggregator endpoints for our perspective outside the container network.
+    let aggregator_endpoints: Vec<_> = leader_task
+        .aggregator_endpoints()
+        .iter()
+        .zip([leader_port, helper_port])
+        .map(|(url, port)| translate_url_for_external_access(url, port))
+        .collect();
+
+    // We generate exactly one batch's worth of measurement uploads to work around an issue in
+    // Daphne at time of writing.
+    let total_measurements: usize = leader_task.min_batch_size().try_into().unwrap();
+
+    match leader_task.vdaf() {
+        task::VdafInstance::Real(VdafInstance::Prio3Aes128Count) => {
+            let vdaf = Prio3::new_aes128_count(2).unwrap();
+
+            let num_nonzero_measurements = total_measurements / 2;
+            let num_zero_measurements = total_measurements - num_nonzero_measurements;
+            assert!(num_nonzero_measurements > 0 && num_zero_measurements > 0);
+            let measurements = iter::repeat(1)
+                .take(num_nonzero_measurements)
+                .interleave(iter::repeat(0).take(num_zero_measurements))
+                .collect::<Vec<_>>();
+
+            submit_measurements_and_verify_aggregate_generic(
+                vdaf,
+                aggregator_endpoints,
+                leader_task,
+                collector_private_key,
+                &measurements,
+                &(),
+                &num_nonzero_measurements.try_into().unwrap(),
+            )
+            .await;
+        }
+        task::VdafInstance::Real(VdafInstance::Prio3Aes128Sum { bits }) => {
+            let vdaf = Prio3::new_aes128_sum(2, *bits).unwrap();
+
+            let measurements = iter::repeat_with(|| (random::<u128>() as u128) >> (128 - bits))
+                .take(total_measurements)
+                .collect::<Vec<_>>();
+
+            submit_measurements_and_verify_aggregate_generic(
+                vdaf,
+                aggregator_endpoints,
+                leader_task,
+                collector_private_key,
+                &measurements,
+                &(),
+                &measurements.iter().sum(),
+            )
+            .await;
+        }
+        task::VdafInstance::Real(VdafInstance::Prio3Aes128Histogram { buckets }) => {
+            let vdaf = Prio3::new_aes128_histogram(2, buckets).unwrap();
+
+            let mut aggregate_result = vec![0; buckets.len() + 1];
```

Comment:
Review note: I'm using `vec![]` because of the following Clippy error. (`Vec::from([0; size])` also wouldn't work here because the size is not known at compile time)

```
error: slow zero-filling initialization
   --> integration_tests/tests/common/mod.rs:210:13
    |
209 |             let mut aggregate_result = Vec::with_capacity(buckets.len() + 1);
    |                                        ------------------------------------- help: consider replace allocation with: `vec![0; buckets.len() + 1]`
210 |             aggregate_result.resize(buckets.len() + 1, 0);
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `-D clippy::slow-vector-initialization` implied by `-D warnings`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#slow_vector_initialization
```

Reply:
Ah, well this is a good enough reason to use `vec!` still. Hopefully the underlying issue can be fixed eventually.